tests/test_fill_sack_from_repos_in_cache.py

```diff
@@ -0,0 +1,191 @@
+    def setUp(self):
+        self.tmpdir = tempfile.mkdtemp(prefix="dnf_test_")
+
+        original_repo_path = os.path.join(os.path.abspath(os.path.dirname(__file__)), "repos/rpm")
+        self.repo_copy_path = os.path.join(self.tmpdir, "repo")
+        shutil.copytree(original_repo_path, self.repo_copy_path)
+
+        self._create_cache_for_repo(self.repo_copy_path, self.tmpdir)
+
+        # Just to be sure remove repo (it shouldn't be used)
+        shutil.rmtree(self.repo_copy_path)
+
+        # Prepare base for the actual test
+        conf = dnf.conf.MainConf()
+        conf.cachedir = os.path.join(self.tmpdir, "cache")
+        self.test_base = dnf.Base(conf=conf)
+        repoconf = dnf.repo.Repo(TEST_REPO_NAME, conf)
+        repoconf.baseurl = self.repo_copy_path
+        repoconf.enable()
+        self.test_base.repos.add(repoconf)
+
+    def tearDown(self):
+        self.test_base.close()
+        shutil.rmtree(self.tmpdir)
+
+    def test_with_solv_solvx_repomd(self):
+        # Remove xml metadata except repomd
+        # repomd.xml is not compressed and doesn't end with .gz
+        repodata_without_repomd = glob.glob(os.path.join(self.tmpdir, "cache/test-tour-repo-*/repodata/*.gz"))
+        for f in repodata_without_repomd:
+            os.remove(f)
+
+        # Now we only have cache with just solv, solvx files and repomd.xml
+
+        self.test_base.fill_sack_from_repos_in_cache(load_system_repo=False)
+
+        q = self.test_base.sack.query()
+        packages = q.run()
+        self.assertEqual(len(packages), 9)
+        self.assertEqual(packages[0].evr, "4-4")
+
+    def test_with_just_solv_repomd(self):
+        # Remove xml metadata except repomd
+        # repomd.xml is not compressed and doesn't end with .gz
+        repodata_without_repomd = glob.glob(os.path.join(self.tmpdir, "cache/test-tour-repo-*/repodata/*.gz"))
+        for f in repodata_without_repomd:
+            os.remove(f)
+
+        # Remove solvx files
+        solvx = glob.glob(os.path.join(self.tmpdir, "cache/*.solvx"))
+        for f in solvx:
+            os.remove(f)
+
+        # Now we only have cache with just solv files and repomd.xml
+
+        self.test_base.fill_sack_from_repos_in_cache(load_system_repo=False)
+
+        q = self.test_base.sack.query()
+        packages = q.run()
+        self.assertEqual(len(packages), 9)
+        self.assertEqual(packages[0].evr, "4-4")
+
+    def test_with_xml_metadata(self):
+        # Remove all solv and solvx files
+        solvx = glob.glob(os.path.join(self.tmpdir, "cache/*.solv*"))
+        for f in solvx:
+            os.remove(f)
+
+        # Now we only have cache with just xml metadata
+
+        self.test_base.fill_sack_from_repos_in_cache(load_system_repo=False)
+
+        q = self.test_base.sack.query()
+        packages = q.run()
+        self.assertEqual(len(packages), 9)
+        self.assertEqual(packages[0].evr, "4-4")
+
+    def test_without_repomd(self):
+        # Remove xml metadata
```

Comment:
Please add a comment that expected behaviour is fail => exception

Reply:
I have added a comment below to all the tests that expect exception.